### PR TITLE
Parse sequences in Rust/wasm

### PR DESCRIPTION
**Pull request by luizirber**
https://github.com/sourmash-bio/sourmash/pull/3047 implemented sequence parsing using `niffler` + `needletail`, which allows FASTA/FASTQ, as well as `gz` or uncompressed files.

This PR are the changes needed to use the package generated from that PR. Once that PR is merged and released it can be used directly here, but wanted to leave this open to allow commenting on the changes I did to make it work on the component.

### CODE REVIEW

**Comment by luizirber** [comment]
I updated A LOT of dependencies, please let me know if I overdid it :sweat_smile:

**Comment by luizirber** [comment]
I added the `wasm-pack` output of https://github.com/sourmash-bio/sourmash/pull/3047 to `pkg/` temporarily to make it easier to verify, but it will obviously be pulled from NPM in the future

**Comment by SandyRogers** [comment]
@luizirber Thank you very much for this PR! @MGS-sails : let's have a chat about this one, I think you were interested in looking into this web assembly piece?

**Comment by MGS-sails** [comment]
> @luizirber Thank you very much for this PR! @MGS-sails : let's have a chat about this one, I think you were interested in looking into this web assembly piece?

@SandyRogers Yes, It'd be great to have a chat about this. 

Thanks !!

**Comment by luizirber** [comment]
Now it uses the released [sourmash 0.13.1](https://www.npmjs.com/package/sourmash/v/0.13.1)

File: src/index.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import style from './index.css';
 
 const worker = new Worker();
 
-const SUPPORTED_EXTENSIONS = ['.fa', '.fasta', '.fna'];
+const SUPPORTED_EXTENSIONS = ['.fa', '.fasta', '.fna', '.gz', '.fq'];
```
Comment (bluegenes) [inline src/index.ts]:
```suggestion
const SUPPORTED_EXTENSIONS = ['.fa', '.fasta', '.fna', '.gz', '.fq', '.fastq'];
```

**Review by MGS-sails — APPROVED** [review]

**Comment by MGS-sails** [comment]
@luizirber sorry for taking so long to get back to you on this.
@SandyRogers the pull request looks good to me. 
I pulled this and tested it locally on our web client and it works as normal. 
The addition of the fastq file support works as expected and I can also see that sketching is happening client side.

**Pull request merged**